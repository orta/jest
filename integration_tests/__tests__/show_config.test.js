/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

'use strict';

const path = require('path');
const skipOnWindows = require('skipOnWindows');
const {linkJestPackage} = require('../utils');
const runJest = require('../runJest');

describe('jest --showConfig', () => {
  skipOnWindows.suite();

  const dir = path.resolve(__dirname, '..', 'verbose_reporter');

  beforeEach(() => {
    if (process.platform !== 'win32') {
      linkJestPackage('babel-jest', dir);
    }
  });

  it('outputs config info and exits', () => {
    const root = path.join(__dirname, '..', '..', '..');
    expect.addSnapshotSerializer({
      print: val =>
        val
          .replace(/"cacheDirectory": "(.+)"/g, '"cacheDirectory": "/tmp/jest"')
          .replace(/"name": "(.+)"/g, '"name": "[md5 hash]"')
          .replace(/"version": "(.+)"/g, '"version": "[version]"')
          .replace(new RegExp(root, 'g'), '/mocked/root/path'),
      test: val => typeof val === 'string',
    });
    const {stdout} = runJest(dir, [
      '--showConfig',
      '--no-cache',
      // Make the snapshot flag stable on CI.
      '--updateSnapshot',
    ]);
    expect(stdout).toMatchSnapshot();
  });
});
