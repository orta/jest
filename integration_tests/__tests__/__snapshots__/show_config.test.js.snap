// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest --showConfig outputs config info and exits 1`] = `
{
  "config": {
    "automock": false,
    "browser": false,
    "cache": false,
    "cacheDirectory": "/tmp/jest",
    "clearMocks": false,
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ],
    "globals": {},
    "haste": {
      "providesModuleNodeModules": []
    },
    "moduleDirectories": [
      "node_modules"
    ],
    "moduleFileExtensions": [
      "js",
      "json",
      "jsx",
      "node"
    ],
    "moduleNameMapper": {},
    "modulePathIgnorePatterns": [],
    "name": "[md5 hash]",
    "resetMocks": false,
    "resetModules": false,
    "rootDir": "/mocked/root/path/jest/integration_tests/verbose_reporter",
    "roots": [
      "/mocked/root/path/jest/integration_tests/verbose_reporter"
    ],
    "setupFiles": [
      "/mocked/root/path/jest/node_modules/regenerator-runtime/runtime.js"
    ],
    "snapshotSerializers": [],
    "testEnvironment": "/mocked/root/path/jest/packages/jest-environment-node/build/index.js",
    "testMatch": [
      "**/__tests__/**/*.js?(x)",
      "**/?(*.)(spec|test).js?(x)"
    ],
    "testPathIgnorePatterns": [
      "/node_modules/"
    ],
    "testRegex": "",
    "testRunner": "/mocked/root/path/jest/packages/jest-jasmine2/build/index.js",
    "testURL": "about:blank",
    "timers": "real",
    "transform": [
      [
        "^.+\\\\.jsx?$",
        "/mocked/root/path/jest/integration_tests/verbose_reporter/node_modules/babel-jest/build/index.js"
      ]
    ],
    "transformIgnorePatterns": [
      "/node_modules/"
    ]
  },
  "framework": "jasmine2",
  "globalConfig": {
    "bail": false,
    "coverageDirectory": "/mocked/root/path/jest/integration_tests/verbose_reporter/coverage",
    "coverageReporters": [
      "json",
      "text",
      "lcov",
      "clover"
    ],
    "expand": false,
    "mapCoverage": false,
    "noStackTrace": false,
    "notify": false,
    "rootDir": "/mocked/root/path/jest/integration_tests/verbose_reporter",
    "testPathPattern": "",
    "testResultsProcessor": null,
    "updateSnapshot": "all",
    "useStderr": false,
    "verbose": null,
    "watch": false,
    "watchman": true
  },
  "version": "[version]"
}

`;
